sklearn/metrics/tests/test_pairwise_distances_reduction.py

```diff
@@ -0,0 +1,466 @@
+import numpy as np
+import pytest
+from collections import defaultdict
+from numpy.testing import assert_array_equal, assert_allclose
+from scipy.sparse import csr_matrix
+
+from sklearn.metrics._dist_metrics import (
+    DenseDenseDatasetsPair,
+)
+
+from sklearn.metrics._pairwise_distances_reduction import (
+    PairwiseDistancesReduction,
+    PairwiseDistancesArgKmin,
+    FastEuclideanPairwiseDistancesArgKmin,
+    _sqeuclidean_row_norms,
+)
+
+from sklearn.utils import _in_unstable_openblas_configuration
+from sklearn.utils.fixes import sp_version, parse_version
+from sklearn.utils._testing import fails_if_unstable_openblas
+
+
+def _get_dummy_metric_params_list(metric: str, n_features: int):
+    """Return list of dummy DistanceMetric kwargs for tests."""
+
+    rng = np.random.RandomState(1)
+    weights = rng.random_sample(n_features)
+    weights /= weights.sum()
+
+    V = rng.random_sample((n_features, n_features))
+
+    # VI is positive-semidefinite, preferred for precision matrix
+    VI = np.dot(V, V.T) + 3 * np.eye(n_features)
+
+    METRICS_PARAMS = defaultdict(
+        list,
+        {
+            "euclidean": [{}],
+            "manhattan": [{}],
+            "minkowski": [dict(p=1.5), dict(p=2), dict(p=3), dict(p=np.inf)],
+            "chebyshev": [{}],
+            "seuclidean": [dict(V=rng.rand(n_features))],
+            "haversine": [{}],
+            "wminkowski": [dict(p=1.5, w=weights)],
+            "mahalanobis": [dict(VI=VI)],
+        },
+    )
+
+    wminkowski_kwargs = dict(p=3, w=rng.rand(n_features))
+
+    if sp_version < parse_version("1.8.0.dev0"):
+        # TODO: remove once we no longer support scipy < 1.8.0.
+        # wminkowski was removed in scipy 1.8.0 but should work for previous
+        # versions.
+        METRICS_PARAMS["wminkowski"].append(wminkowski_kwargs)  # type: ignore
+    else:
+        # Recent scipy versions accept weights in the Minkowski metric directly:
+        # type: ignore
+        METRICS_PARAMS["minkowski"].append(wminkowski_kwargs)  # type: ignore
+
+    return METRICS_PARAMS.get(metric, [{}])
+
+
+def assert_radius_neighborhood_results_equality(ref_dist, dist, ref_indices, indices):
```

Comment:
We can remove this since we are not using `assert_radius_neighborhood_results_equality` for this PR.